test/widgets/message_list_test.dart

```diff
@@ -576,6 +577,107 @@ void main() {
+      check(find.byIcon(ZulipIcons.message_moved).evaluate()).length.equals(moved);
+    }
+
+    testWidgets('no edited or moved messages', (WidgetTester tester) async {
+      final message = eg.streamMessage();
+      await setupMessageListPage(tester, messages: [message]);
+      checkMarkersCount(edited: 0, moved: 0);
+    });
+
+    testWidgets('edited and moved messages from events', (WidgetTester tester) async {
+      final message = eg.streamMessage();
+      final message2 = eg.streamMessage();
+      await setupMessageListPage(tester, messages: [message, message2]);
+      checkMarkersCount(edited: 0, moved: 0);
+
+      await store.handleEvent(eg.updateMessageEditEvent(message, renderedContent: 'edited'));
+      await tester.pump();
+      checkMarkersCount(edited: 1, moved: 0);
+
+      await store.handleEvent(eg.updateMessageMoveEvent(
+        [message, message2], origTopic: 'old', newTopic: 'new'));
+      await tester.pump();
+      checkMarkersCount(edited: 1, moved: 1);
+
+      await store.handleEvent(eg.updateMessageEditEvent(message2, renderedContent: 'edited'));
+      await tester.pump();
+      checkMarkersCount(edited: 2, moved: 0);
+    });
+
+    List<Map<String, List<Rect>>> captureMessageRects(
+      WidgetTester tester,
+      List<Message> messages,
+      Message targetMessage,
+    ) {
+      assert(messages.contains(targetMessage));
+      final List<Map<String, List<Rect>>> result = [];
+      for (final message in messages) {
+        final baseFinder = find.byWidgetPredicate(
+          (widget) => widget is MessageWithPossibleSender
+            && widget.item.message.id == message.id);
+
+        List<Rect> getRectsMatching(Finder matching) {
+          final finder = find.descendant(
+            of: baseFinder, matching: matching)..tryEvaluate();
+          check(finder.found, because: '${message.content}: $matching')
+            .isNotEmpty();
+          return List.generate(
+            finder.found.length,
+            (i) => tester.getRect(finder.at(i)));
+        }
+
+        result.add({
+          'MessageWithPossibleSender': [tester.getRect(baseFinder)],
+          'MessageContent': getRectsMatching(find.byType(MessageContent)),
+          'Paragraph': getRectsMatching(find.byType(Paragraph)),
+          if (message.id == targetMessage.id) ...{
+            'MessageImage': getRectsMatching(find.byType(MessageImage)),
+            'Star': getRectsMatching(find.byIcon(ZulipIcons.star_filled)),
+            'ReactionChip': getRectsMatching(find.byType(ReactionChip)),
+            'ReactionChipsList': getRectsMatching(find.byType(ReactionChipsList)),
+          },
+        });
+      }
+      return result;
+    }
+
+    testWidgets('edit state updates do not affect layout', (WidgetTester tester) async {
+      prepareBoringImageHttpClient();
+      final messages = [
+        eg.streamMessage(),
+        eg.streamMessage(
+          content: ContentExample.imageClusterThenContent.html,
```

Comment:
I wanted to test that because I discovered the part with `MessageImageList` do not respond to swipe on the animated branch. It is not relevant now but might be later.

Reply:
This is something I should note in the commit message, apparently.

Reply:
Interesting, that makes sense.

I think that still won't be relevant for this layout test, though — having the image interfere with a gesture (because the image has its own gesture recognizers) is a lot more natural than having it interfere with an ancestor's layout. So it's still probably best left out of this test, but then you might use this example in a new test in the swipe-to-reveal PR.

Reply:
(Or even in that later test, it sounds like `imageSingle` would likely be enough to make the point — it has an image and even a paragraph too, just one of each instead of two of each.)